### ISSUE TITLE
Flags 'On Error GoTo -1' as a NegativeLineNumber

### ISSUE THREAD

**Issue by DavidS-BDO** [rubberduck-vba/Rubberduck#5682]
**Rubberduck version information**
  Version 2.5.1.5557
  OS: Microsoft Windows NT 10.0.18363.0, x64
  Host Product: Microsoft Office x64
  Host Version: 16.0.12527.21594
  Host Executable: EXCEL.EXE

**Description**
  The Code inspection flags  'On Error GoTo -1' as a NegativeLineNumber. 
  'On Error GoTo -1' is typically the best way to clear errors after they are handled. 

**Expected behavior**
  It should not flag these as an error.

**Additional context**
https://www.automateexcel.com/vba/error-handling/#ErrClearOn_Error_GoTo_-1_and_Resetting_ErrNumber

> Sub ErrExamples()
>
>    On Error GoTo errHandler:
>       
>    Error (13)                                   '"Type mismatch" error
>    
>    Exit Sub
>errHandler:
>    ' Clear Error. To test swap the 'On Error GoTo -1' and'Err.Clear'
>    Debug.Print err.Number, err.Description
>    On Error GoTo -1
>    'Err.Clear
>    Debug.Print err.Number, err.Description
>    On Error GoTo errHandler2:
>
>    Error (1034)                                 '"Application-defined" error
> 
>    Exit Sub
>errHandler2:
>    Debug.Print err.Number, err.Description
>End Sub

**Comment by bclothier** [comment]
While I agree this is a bug, it's not the bug you think it is. 

First, it is entirely intentional that the code with `On Error GoTo -1` get flagged. However, it shouldn't have been the negative line number inspection that gets flagged. Instead, it should have been the On Error GoTo -1 inspection that gets flagged. 

We actually want to catch both cases of negative line number and the `On Error GoTo -1` because those are quite obscure and undocumented. The negative line number is likely an unintentional design of VBA. In the case of `On Error GoTo -1`, we still flag them because as far as I know, Microsoft has not actually documented this. For example, [Error Statement](https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/on-error-statement) does not describe this. If you can point to a official Microsoft reference that does that, I'd love to know about it. 

But even if it were documented, the argument is that if you need multiple error handling, your procedure is likely doing too much anyway! It's the same flaw as using `GoSub...Return` and we want to encourage clear and readable code by encouraging to split those into multiple procedures. Even if the new procedure is only called by one other procedure, the ability to provide meaningful names to procedure will make it much more clear than writing multiple error handling and then having to add comments to explain why the multiple error handling. 

Contrast:

```
Public Sub DoIt()
  On Error GoTo ErrHandler1
  Dim FSO As Scripting.FileSystemObject
  Set FSO = New Scripting.FileSystemObject
  
  Dim MyTextFile As Scripting.TextStream
  MyTextFile = FSO.OpenTextFile("C:\foo\bar.txt")
  
  On Error GoTo ErrHandler2
  Dim i As Long
  For i = 1 To 100
    MyTextFile.WriteLine "I will not pull my classmate's pigtails again."
  Next

ExitProc:
  On Error Resume Next
  MyTextFile.Close
  On Error GoTo 0

ErrHandler2:
  MsgBox "Oops, I pulled her pigtails! He, he, he!"
  Resume ExitProc

ErrHandler1:
  MsgBox "Welp, I can't write it 100 times... I'm outta here! Peace out!"
  Resume ExitProc
End Sub
```

```
Public Sub DoIt()
  On Error GoTo ErrHandler

  Dim MyTextFile As Scripting.TextStream
  If TryGetTextFile("C:\foo\bar.txt", MyTextFile) Then
    Dim i As Long
    For i = 1 To 100
       MyTextFile.WriteLine "I will not pull my classmate's pigtails again."
    Next
  Else
    MsgBox "Welp, I can't write it 100 times... I'm outta here! Peace out!"
  End If

ExitProc:
  On Error Resume Next
  MyTextFile.Close
  On Error GoTo 0
  Exit Sub

ErrHandler:
  MsgBox "Oops, I pulled her pigtails! He, he, he!"
  Resume ExitProc
End Sub

Private Function TryGetTextFile( _
  ByVal FilePath As String, 
  ByRef OutTextFile As Scripting.TextStream _
) As Boolean
  Dim FSO As Scripting.FileSystemObject
  Set FSO = New Scripting.FileSystemObject

  On Error Resume Next
  Set OutTextFile = FSO.CreateTextFile(FilePath)
  On Error GoTo 0

  TryGetTextFile = Not (OutTextFile Is Nothing)
End Function
```

Hopefully, you will see that by splitting out the logical unit of creating and getting a handle on a text file from the procedure, we've simplified the error handling requirement considerably. More importantly, the modified version is easier to read -- it's much easier to digest `TryGetTextFile` than it is to digest several discrete statements and a out of flow error handler at the bottom procedure. Therefore, if one follows that pattern, I think it will be exceedingly hard if not impossible to find a legitimate situation where you absolutely must use `On Error GoTo -1` just to reset the error state. Even in the case where your error handler needed to do something that potentially could error (example being writing a log file but you never know whether there's room on the hard drive), it's still better off splitting it into own procedure which will then have its own error state.

**Comment by MDoerner** [comment]
Based on the info you provide, you use the stable release version. This bug has already been closed in PR #5594, which is part of a later prerelease.

It is basically a duplicate of #5544.
